src/analysis/functions.rs

```diff
@@ -188,6 +188,67 @@ fn analyze_function(
     );
     parameters.analyze_return(env, &ret.parameter);
 
+    // Workaround for some _hash() / _compare() / _equal() functions taking
+    // "gconstpointer" as arguments instead of the actual type
```

Comment:
This is all a bit ugly (should be moved to a separate function too) but the main problem is that we use `to_glib_none().0` but the FFI functions expects a `*const c_void` and not a `*mut WhateverFFIType`.

@EPashkin How would we solve this best? Apart from that this is working fine

Reply:
IMHO this solution fine if separated to function,
other option is replace type parameter on preprocess stage if mode normal (or in sys too to remove "*const void" from member functions)

Reply:
The problem is how to to make the FFI function accept the right type, or how to add the correct cast after `to_glib_none().0`.

Would you prefer to have those functions fixed up in FFI to take the actual type instead of gconstpointer?

Reply:
I thought that `to_glib_extra` help with ` as *const _` but I was wrong.
Currently I don't see way to add something after `.to_glib_none().0"`.
Fixing in FFI IMHO removed uncertainness  about parameter meaning,
but it will don't works good for traited `compare`, so also not solution.
So actual solution seems add analog `to_glib_extra` to `TransformationType::ToGlibPointer` and fill it here

Reply:
ok thanks :)

Reply:
Works for GLib but not for `GObjects`

```
error[E0282]: type annotations needed
   --> src/auto/file.rs:816:30
    |
816 |             ffi::g_file_hash(self.to_glib_none().0 as glib_ffi::gconstpointer)
    |                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ cannot infer type for `P`
```

Need to call a specific `to_glib_none()` here for the target type. Do we have infrastructure for that?

Reply:
We don't have, I even can't write oneline here that return right type.
Seems for function in trait we need use 2 transformation:
TransformationType::ToGlibStash, 
new transformation that was part of TransformationType::Into for with_stash = true
Last need changed too to allow define type.
Name for new transformation IMHO better be "ToGlibStash" with old renamed to "UseGlibStash"
Problem with name of this variable, and I can't get working code :(
Last try
```
    fn hash(&self) -> u32 {
        unsafe {
            let this: Stash<*mut ffi::GFileIface, _> = self.to_glib_none();
            ffi::g_file_hash(this.0 as *const _)
        }
    }
```

Reply:
```rust
    fn hash(&self) -> u32 {
        unsafe {
            ffi::g_file_hash(ToGlibPtr::<*mut ffi::GFile>::to_glib_none(self).0 as glib_ffi::gconstpointer)
        }
    }
```

This works. I guess that can relatively easily be special-cased

Reply:
@EPashkin How do I get a string of the FFI type name (`*mut ffi::GFile`) from a type ID in GIR?

Reply:
You need add it too, all function in ffi_type.rs need "inner" string.
Use `Type::glib_get_name()` https://github.com/gtk-rs/gir/blob/master/src/analysis/ffi_type.rs#L126
and `fix_name` here.

Reply:
implemented that now and it seems to work. To the next step! :)

Reply:
Feel free to review the existing commits already btw, everything else is only on top of this